devito/ir/clusters/cluster.py

```diff
@@ -230,7 +230,18 @@ def is_dense(self):
 
     @property
     def is_sparse(self):
-        return not self.is_dense
+        """
+        A cluster is sparse if it represent a sparse operation i.e if both
+
+            * The cluster contains sparse functions
+            * The cluster uses dense functions
+
+        If only the first case is true, the cluster only contains operation on the sparse
+        function itself without indirection and therefore only contains dense operations.
+        """
+        return (any(f.is_SparseFunction for f in self.functions) and
+                len([f for f in self.functions
```

Comment:
what if an Array?

perhaps a bit safer:

```
return all(split(self.functions, lambda f: f.is_SparseFunction))
```

?

Reply:
Not sure that's enough, that doesn't check if there actually is a sparse function I'll see if can simplify

Reply:
ah OK